build/references/command-line-interface.md

```diff
@@ -254,65 +254,165 @@ Enables peer specific query latency metrics. Defaults to `false`.
 
 Minimum amount of time messages to a peer must be failing before the peer is benched. Defaults to `5m`.
 
-### C-Chain / Coreth
+### Chain Configs
 
-`--coreth-config` \(json\):
+Some chains (right now, just the C-Chain) allow the node operator to provide a custom configuration. AvalancheGo can read chain configurations from files and pass them to the corresponding chains on initialization.
 
-This allows you to specify a config to be passed into the C-Chain. The default values for this config are:
+AvalancheGo looks for these files in the directory specified by `--chain-config-dir`. This directory can have sub-directories whose names are chain IDs or chain aliases. Each sub-directory contains the configuration for the chain specified in the directory name. Each sub-directory should contain a file named `config`, whose value is passed in when the corresponding chain is initialized. For example, the config for the C-Chain should be at: `[chain-config-dir-goes-here]/C/config.json`.
+
+The extension that these files should have, and the contents of thse files, is VM-dependent. For example, some chains may expect `config.txt` while others expect `config.json`. If multiple files are provided with the same name but different extensions (e.g. `config.json` and `config.txt`) in the same sub-directory, AvalancheGo will exit with an error.
+
+For a given chain, AvalancheGo will look first for a config sub-directory whose name is the chain ID. If it isn't found, it looks for a config sub-directory whose name is the chain's primary alias. If it's not found, it looks for a config sub-directory whose name is another alias for the chain. All folder and file names are case sensitive.
+
+It is not required to provide these custom configurations. If they are not provided, a VM-specific default config will be used.
+
+`--chain-config-dir` \(string\):
+
+Specifies the directory that contains chain configs, as described above. Defaults to `$HOME/.avalanchego/configs/chains`. If this flag is not provided and the default directory does not exist, AvalancheGo will not exit since custom configs are optional. However, if the flag is set, the specified folder must exist, or AvalancheGo will exit with an error.
+
+#### C-Chain Configs
 
-```cpp
+Currently, the C-Chain is the only chain that supports custom configurations. In order to specify config for the C-Chain, a JSON config file should be put at `{chain-config-dir}/C/config.json` (or another valid location, as specified above.)
```

Comment:
In order to specify a config for the C-Chain, a JSON config file should be placed at